### ISSUE TITLE
False positive UnusedVariable when using ++$h as direct input

### ISSUE THREAD

**Issue by 7ochem** [vimeo/psalm#6712]
In this example 

```php
$h = 0;

echo sprintf("This is %d.\n", ++$h);
echo sprintf("And then this is %d.\n", ++$h);
echo sprintf("And finally comes %d.\n", ++$h);
```
Psalm flags `$h` to be unused in the last line it is being used.

https://psalm.dev/r/b19073159b

On psalm.dev this leads to a `INFO`. In my config is leads to an `ERROR`:

```
ERROR: UnusedVariable - src/Foo.php - $h is never referenced or the value is not used (see https://psalm.dev/077)
    echo sprintf('%d. Something something...', ++$h);
```

Config:
```xml
<psalm
    errorLevel="4"
    findUnusedVariablesAndParams="true"
```

On vimeo/psalm 4.10.0
